internal/framework/status/updater.go

```diff
@@ -169,3 +182,43 @@ func (upd *updaterImpl) update(
 			"kind", obj.GetObjectKind().GroupVersionKind().Kind)
 	}
 }
+
+func UpdateObjStatus(
+	ctx context.Context,
+	attempts int,
+	sleep time.Duration,
+	obj client.Object,
+	upd *UpdaterImpl,
+) (err error) {
+	for i := 0; i < attempts; i++ {
+		if i > 0 {
+			time.Sleep(sleep)
+			sleep *= 2
+		}
+		if err = upd.cfg.Client.Status().Update(ctx, obj); err == nil {
+			return nil
+		}
+	}
+	return err
+}
+
+func GetObj(
```

Comment:
Since GetObj and UpdateObj aren't used outside the package I don't think we need to export them. If you want to test them you can add a test file with the same package name so you don't have to import them.

Reply:
Removed because we use imported retrying function.